From 98e9b7636d9a89640fba3149f9675d3a867c251b Mon Sep 17 00:00:00 2001
From: Oleksandr Dmytryshyn <oleksandr.dmytryshyn@globallogic.com>
Date: Thu, 3 Sep 2015 19:05:45 +0300
Subject: [PATCH 1/5] domctl: introduce set_11_mapping

The following call is designed to check if domain
should be mapped 1:1 or not.

Change-Id: I8dfec2fb93fec99519478707105a61472a213adf
Signed-off-by: Oleksandr Dmytryshyn <oleksandr.dmytryshyn@globallogic.com>
Signed-off-by: Iurii Konovalenko <iurii.konovalenko@globallogic.com>
---
 xen/include/public/domctl.h         | 2 ++
 xen/xsm/flask/hooks.c               | 3 +++
 xen/xsm/flask/policy/access_vectors | 4 ++++
 3 files changed, 9 insertions(+)

diff --git a/xen/include/public/domctl.h b/xen/include/public/domctl.h
index 2457698..70bcb74 100644
--- a/xen/include/public/domctl.h
+++ b/xen/include/public/domctl.h
@@ -1211,6 +1211,8 @@ struct xen_domctl {
 #define XEN_DOMCTL_monitor_op                    77
 #define XEN_DOMCTL_psr_cat_op                    78
 #define XEN_DOMCTL_soft_reset                    79
+#define XEN_DOMCTL_access_remote_pagedir         80
+#define XEN_DOMCTL_set_11_mapping                81
 #define XEN_DOMCTL_gdbsx_guestmemio            1000
 #define XEN_DOMCTL_gdbsx_pausevcpu             1001
 #define XEN_DOMCTL_gdbsx_unpausevcpu           1002
diff --git a/xen/xsm/flask/hooks.c b/xen/xsm/flask/hooks.c
index 20d46c8..e7044d0 100644
--- a/xen/xsm/flask/hooks.c
+++ b/xen/xsm/flask/hooks.c
@@ -742,6 +742,9 @@ static int flask_domctl(struct domain *d, int cmd)
     case XEN_DOMCTL_soft_reset:
         return current_has_perm(d, SECCLASS_DOMAIN2, DOMAIN2__SOFT_RESET);
 
+    case XEN_DOMCTL_set_11_mapping:
+        return current_has_perm(d, SECCLASS_DOMAIN2, DOMAIN2__SET_11_MAPPING);
+
     default:
         printk("flask_domctl: Unknown op %d\n", cmd);
         return -EPERM;
diff --git a/xen/xsm/flask/policy/access_vectors b/xen/xsm/flask/policy/access_vectors
index 3d29042..7ce26d6 100644
--- a/xen/xsm/flask/policy/access_vectors
+++ b/xen/xsm/flask/policy/access_vectors
@@ -248,6 +248,10 @@ class domain2
     mem_sharing
 # XEN_DOMCTL_psr_cat_op
     psr_cat_op
+# XEN_DOMCTL_access_remote_pagedir
+    access_remote_pagedir
+# XEN_DOMCTL_set_11_mapping
+    set_11_mapping
 }
 
 # Similar to class domain, but primarily contains domctls related to HVM domains
-- 
1.9.1

